pkg/evm/evm.go

```diff
@@ -0,0 +1,183 @@
+	client ethclient.Client,
+	contractAddressStr string,
+) ([]byte, error) {
+	ctx, cancel := utils.GetAPIContext()
+	defer cancel()
+	contractAddress := common.HexToAddress(contractAddressStr)
+	return client.CodeAt(ctx, contractAddress, nil)
+}
+
+func GetAddressBalance(
+	client ethclient.Client,
+	addressStr string,
+) (*big.Int, error) {
+	ctx, cancel := utils.GetAPIContext()
+	defer cancel()
+	address := common.HexToAddress(addressStr)
+	return client.BalanceAt(ctx, address, nil)
+}
+
+// Returns the gasFeeCap, gasTipCap, and nonce the be used when constructing a transaction from address
+func CalculateTxParams(
+	client ethclient.Client,
+	addressStr string,
+) (*big.Int, *big.Int, uint64, error) {
+	ctx, cancel := utils.GetAPIContext()
+	defer cancel()
+	address := common.HexToAddress(addressStr)
+	baseFee, err := client.EstimateBaseFee(ctx)
+	if err != nil {
+		return nil, nil, 0, err
+	}
+	gasTipCap, err := client.SuggestGasTipCap(ctx)
+	if err != nil {
+		return nil, nil, 0, err
+	}
+	nonce, err := client.NonceAt(ctx, address, nil)
+	if err != nil {
+		return nil, nil, 0, err
+	}
+	gasFeeCap := baseFee.Mul(baseFee, big.NewInt(BaseFeeFactor))
+	gasFeeCap.Add(gasFeeCap, big.NewInt(MaxPriorityFeePerGas))
+	return gasFeeCap, gasTipCap, nonce, nil
+}
+
+func FundAddress(
+	client ethclient.Client,
+	sourceAddressPrivateKeyStr string,
+	targetAddressStr string,
+	amount *big.Int,
+) error {
+	sourceAddressPrivateKey, err := crypto.HexToECDSA(sourceAddressPrivateKeyStr)
+	if err != nil {
+		return err
+	}
+	sourceAddress := crypto.PubkeyToAddress(sourceAddressPrivateKey.PublicKey)
+	gasFeeCap, gasTipCap, nonce, err := CalculateTxParams(client, sourceAddress.Hex())
+	if err != nil {
+		return err
+	}
+	targetAddress := common.HexToAddress(targetAddressStr)
+	ctx, cancel := utils.GetAPIContext()
+	defer cancel()
+	chainID, err := client.ChainID(ctx)
+	if err != nil {
+		return err
+	}
+	tx := types.NewTx(&types.DynamicFeeTx{
+		ChainID:   chainID,
+		Nonce:     nonce,
+		To:        &targetAddress,
+		Gas:       NativeTransferGas,
+		GasFeeCap: gasFeeCap,
+		GasTipCap: gasTipCap,
+		Value:     amount,
+	})
+	txSigner := types.LatestSignerForChainID(chainID)
+	signedTx, err := types.SignTx(tx, txSigner, sourceAddressPrivateKey)
+	if err != nil {
+		return err
+	}
+	if err := client.SendTransaction(ctx, signedTx); err != nil {
+		return err
+	}
+	_, b, err := WaitForTransaction(client, signedTx)
+	if err != nil {
+		return err
+	}
+	if !b {
+		return fmt.Errorf("failure funding %s from %s amount %d", targetAddressStr, sourceAddress.Hex(), amount)
+	}
+	return nil
+}
+
+func IssueTx(
+	client ethclient.Client,
+	txStr string,
+) error {
+	tx := new(types.Transaction)
+	if err := tx.UnmarshalBinary(common.FromHex(txStr)); err != nil {
+		return err
+	}
+	ctx, cancel := utils.GetAPIContext()
+	defer cancel()
+	if err := client.SendTransaction(ctx, tx); err != nil {
+		return err
+	}
+	_, b, err := WaitForTransaction(client, tx)
```

Comment:
can `WaitForTransaction(client,tx) ` be replaced with `client.WaitForTransaction(tx)` to be consistent with ` client.SendTransaction`